### ISSUE TITLE
Adding taxonomy="true" with no taxonomy terms present in URL causes fatal error

### ISSUE THREAD

**Issue by ldstevens** [statamic/v2-hub#381]
I have the following template for my home page:

```
{{ collection:blog as="entries" }}
{{ entries }}

    {{ partial:blog-list }}

{{ /entries }}
{{ /collection:blog }}
```

I would like to add `taxonomy="true"` to those tags so a URL e.g. /tags/my-tag filters the list accordingly. In slack Jack mentioned this should be possible; however currently adding taxonomy="true" causes the following error:

```
Whoops, looks like something went wrong.

1/1
FatalErrorException in CollectionTags.php line 164:
Call to a member function id() on a non-object
in CollectionTags.php line 164
```

... hence this issue :) The error persists if I remove as="entries" {{ entries }}{{ /entries }} too (making the template look identical to the one below).

I have worked around this by creating another template:

```
{{ collection:blog taxonomy="true" }}

{{ partial:blog-list }}

{{ /collection:blog }}
```

...using routes to send requests to it, e.g.:

```
taxonomies:
  tags: '/tags/{slug}'
```

... and that works, fwiw :)

**Comment by edalzell** [comment]
@ldstevens is this still an issue?

**Comment by jackmcdade** [comment]
This one was fixed in 2.5.3. 👍
